OpenRA.Mods.Common/Traits/Air/Aircraft.cs

```diff
@@ -1088,6 +1089,109 @@ public void ResolveOrder(Actor self, Order order)
 
 		#endregion
 
+		bool CanOrder(Actor actor, CPos cell)
+		{
+			if (actor == null || actor.IsDead)
+				return false;
+
+			var aircraft = actor.TraitOrDefault<Aircraft>();
+			return aircraft != null && (aircraft.Info.MoveIntoShroud || self.Owner.Shroud.IsExplored(cell));
+		}
+
+		public IEnumerable<Pair<Actor, CPos>> GroupMove(CPos cell, Order order)
```

Comment:
Thinking ahead: Would it be more effort than its worth to make this generic enough to use with `Mobile` too?

Reply:
It was certainly my aim to apply this to ground units too eventually, but not in the short term. The problem with ground units is that you also have to deform the formation to fit the terrain. e.g., if you order your units to the edge of a cliff, you wouldn't want half your units to go in a completely different direction to get to the other side of the cliff (even worse if the other side isn't accessible at all).

This requires some more conceptualization before we can get to coding and meanwhile air units are relatively simple to do and there is more immediate need for it given things like #17293.

Reply:
I see. In principle there is a relatively simple solution to the terrain issue that shouldn't be *too* painful perf-wise for small-to-reasonable numbers of units: avoid `FindTilesInCircle`, and instead grow the area by taking successive groups of neighbouring cells starting at the target location. Impassible cells would be dropped at each step, so we wouldn't grow past any impassible barriers. The one tricky bit is that we would need to this calculation separately for each locomotor type (consider: hover units or groups of mixed ground/naval).